WordPress/Classes/Services/ReaderSiteService.m

```diff
@@ -149,12 +149,12 @@ - (void)unfollowSiteAtURL:(NSString *)siteURL success:(void(^)(void))success fai
 
 - (void)syncPostsForFollowedSites
 {
-    ReaderAbstractTopic *followedSites = [ReaderAbstractTopic lookupFollowedSitesTopicInContext:self.managedObjectContext];
-    if (!followedSites) {
-        return;
-    }
-
     [[ContextManager sharedInstance] performAndSaveUsingBlock:^(NSManagedObjectContext *context) {
+        ReaderAbstractTopic *followedSites = [ReaderAbstractTopic lookupFollowedSitesTopicInContext:context];
+        if (!followedSites) {
```

Comment:
"followed sites" made me want to suggest to return also if the count is 0, but that's not how the `ReaderAbstractTopic` works.